webapp/packages/core-events/src/NotificationService.ts

```diff
@@ -17,46 +17,63 @@ import {
 
 @injectable()
 export class NotificationService {
-  readonly notificationList = new OrderedMap<number, INotification<any>>(({ id }) => id);
+  readonly notificationList = new OrderedMap<number, INotification<any, any>>(({ id }) => id);
   private notificationNextId = 0
+  private maxPersistentAllow = 1
 
   constructor(
     private settings: EventsSettingsService,
   ) {}
 
-  notify<T = never>(options: INotificationOptions<T>, type: ENotificationType) {
+  notify<T = never, TProps = Record<string, any>>(options: INotificationOptions<T, TProps>, type: ENotificationType) {
     const id = this.notificationNextId++;
 
-    const notification: INotification<T> = {
+    const notification: INotification<T, TProps> = {
       id,
       title: options.title,
       message: options.message,
       details: options.details,
       isSilent: !!options.isSilent,
       customComponent: options.customComponent,
+      extraProps: options.extraProps || {} as TProps,
       source: options.source!,
+      persistent: options.persistent,
       type,
       close: this.close.bind(this, id),
       showDetails: this.showDetails.bind(this, id),
     };
+    const persistent = this.notificationList.values.filter(value => value.persistent);
+    const persistentQty = persistent.length;
+    if (persistentQty >= this.maxPersistentAllow && notification.persistent) {
+      return;
+    }
     this.notificationList.addValue(notification);
 
-    if (this.notificationList.values.length > this.settings.settings.getValue('notificationsPool')) {
-      this.notificationList.remove(this.notificationList.keys[0]);
+    const filteredNotificationList = this.notificationList.values.filter(notification => !notification.persistent);
+
+    if (filteredNotificationList.length > this.settings.settings.getValue('notificationsPool')) {
+      let i = 0;
+      while (this.notificationList.get(this.notificationList.keys[i])?.persistent) {
+        i++;
+      }
+      this.notificationList.remove(this.notificationList.keys[i]);
     }
   }
 
-  customNotification<T = never>(
-    component: () => NotificationComponent<T>,
+  customNotification<T = never, TProps = Record<string, any>>(
```

Comment:
function is too complicated we should move the `source` to base `TProps` interface, also we should add the possibility to override the notification type by the `options` parameter (probably better to use a union for this function)